xea-web/src/components/login/LogInAndOut.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { Component, PropTypes } from 'react'
+import { connect } from 'react-redux'
+import { Modal } from 'react-bootstrap'
+import { loginRequest, logoutRequest } from '../../actions/authActions'
+
+import LoginForm from './LoginForm'
+import UserMenu from './UserMenu'
+
+class LogInAndOut extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = { showModal: false }
+
+    this.open = this.open.bind(this)
+    this.close = this.close.bind(this)
```

Comment:
Nunca vira esta forma de facer as cousas. Non sería máis sencillo usar propiedades da clase de esta outra forma?

```javascript
class LogInAndOut extends Component {
  open = () => { this.setState({showModal: true}) }

  // ...
}
```